src/lib/Libutil/misc_utils.c

```diff
@@ -1363,3 +1363,60 @@ get_mem_info(void) {
 }
 #endif /* malloc_info */
 #endif /* WIN32 */
+
+/**
+ * @brief
+ *	Return a copy of 'str' where non-printing characters
+ *	(except the ones listed in the local variable 'special_char') are
+ *	shown in ^<translated_char> notation.
+ *
+ * @param[in]	str - input string
+ *
+ * @return char *
+ * @retval NULL - Error
+ *
+ * @note
+ * 	Do not free the return value - it's in a fixed memory area that
+ *	will get overwritten the next time the function is called.
+ *      So best to use the result immediately or strdup() it.
+ */
+char *
+show_nonprint_chars(char *str)
+{
+#ifndef WIN32
+	static char	*locbuf = NULL;
+	static size_t	locbuf_size = 0;
+	char		*buf, *buf2;
+	size_t		nsize;
+	int		ch;
+	char		special_char[] = "\n\t";
+
+	if ((str == NULL) || (str[0] == '\0'))
+		return str;
+
+	nsize = (strlen(str) * 2) + 1;
+	if (nsize > locbuf_size) {
+		char *tmpbuf;
+		if ((tmpbuf = realloc(locbuf, nsize)) == NULL)
+			return NULL;
```

Comment:
Won't printf() have a problem if NULL is returned? Should we return a static string instead?

Reply:
Good point. Better yet, I'll just return the original string as is, meaning translation not possible at this time due to realloc() error.